classes/local/groups.php

```diff
@@ -68,16 +68,34 @@ public function __construct($apirest = null, $rand = null) {
      * @return array[stdClass] $groups
      */
     public function get_groups() {
-        $response = $this->apirest->get_groups(10000, 1);
-        if (!isset($response->groups)) {
-            throw new \moodle_exception('getgroupserror', 'accredible', 'https://help.accredible.com/hc/en-us');
-        }
+        $pagesize = 50;
+        $page = 1;
+
+        // Maximum number of pages to request to avoid possible infinite loop.
+        $looplimit = 100;
```

Comment:
Question - I assume you're following the same logic as the `local/credentials.get_credentials`, but I am wondering if this `$looplimit` can be redundant.